packages/map-template/src/components/RouteInstructions/RouteInstructions.jsx

```diff
@@ -18,22 +23,64 @@ function RouteInstructions({ steps, onNextStep, onPreviousStep, originLocation }
     /** Referencing the previous step of each active step */
     const [previous, setPrevious] = useState();
 
-    const [activeStep, setActiveStep] = useState(0);
+    const [activeStep, setActiveStep] = useRecoilState(activeStepState);
+    const [totalSteps, setTotalSteps] = useState();
 
-    /*
-     * Make sure to reset active step whenever the steps change.
+    const [lastStepZoom, setLastStepZoom] = useState();
+    const [lastStepCenter, setLastStepCenter] = useState();
```

Comment:
The two useState could be merged into one (object) for performance reasons. We always read and set both of them.

Reply:
✅